native/PluginProcessor.cpp

```diff
@@ -262,9 +262,23 @@ void EffectsPluginProcessor::initJavaScriptEngine()
         return choc::value::Value();
     });
 
-    jsContext.registerFunction("__log__", [](choc::javascript::ArgumentList args) {
+    jsContext.registerFunction("__log__", [&](choc::javascript::ArgumentList args) {
+
+        const auto* kDispatchScript = R"script(
+(function() {
+  console.log(%);
+  return true;
+})();
+)script";
+
         for (size_t i = 0; i < args.numArgs; ++i) {
             DBG(choc::json::toString(*args[i], true));
+
+            // Dispatch to the UI if it's available
+            if (auto* editor = static_cast<WebViewEditor*>(getActiveEditor())) {
```

Comment:
The general idea here is perfect, but I think we should change the implementation slightly– right now, this will evaluate a new expression in the webView for each argument passed to the `__log__` function. I think instead of running the `for` loop over `args.numArgs` we should serialize `args` as its own array, and then change the `console.log(%)` expression to `console.log(...JSON.parse(%))` so that we only evaluate the expression once, but we still pass the var args to `console.log` correctly

Reply:
Sounds good!